src/test/scala/scala/collection/immutable/TestLazyListExtensions.scala

```diff
@@ -0,0 +1,115 @@
+/*
+ * Scala (https://www.scala-lang.org)
+ *
+ * Copyright EPFL and Lightbend, Inc.
+ *
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package scala.collection.immutable
+
+import org.junit.Assert._
+import org.junit.Test
+
+import next._
+
+class TestLazyListExtensions {
+
+  // This method will *not* terminate for non-cyclic infinite-sized collections.
+  def assertConstantMemory[T](xs: LazyList[T]): Unit =
+    // `force` does cycle detection, so if this terminates, the collection is
```

Comment:
is there something in sbt that will stop this if it's wrong? I'm not super keen on the fail state of a test being hanging

Reply:
well, I'm not keen on it either... I'll add this comment:

```
  // (It's kind of nasty to have tests whose failure mode is to hang, but I don't                                        
  // see an obvious alternative that doesn't involve copying code from LazyList.                                        
  // Perhaps this could be improved at the time this all gets merged into stdlib.)                                      
```